tools/initialize.py

```diff
@@ -0,0 +1,404 @@
+        end = timer()
+
+        print('Rank: ', rank, ' Time to send Edges to proc: ', part_id, ' is : ', timedelta(seconds = end - start))
+
+
+def send_node_features(rank, node_data, node_features, rank_list, ntype_map):
+    """
+    Function to send node_features data to non-rank-0 processes.
+
+    Parameters:
+    -----------
+    rank : rank of the process
+    node_features : node_features, data from the node_feats.dgl
+    ranks_list : list of unique ranks/partition-ids
+    ntype_map : dictionary of mappings between ntype_name -> ntype_id
+    """
+
+    node_features_rank_lst = []
+    for part_id in rank_list: 
+        if part_id == rank: 
+            node_features_rank_lst.append( None )
+            continue
+
+        send_node_features = {}
+        for x in ntype_map.items(): 
+            node_type_name = x[0]
+            node_type_id = x[1]
+            
+            #extract orig_type_node_id
+            idx = (node_data[:,7] == part_id) & (node_data[:,0] == node_type_id) 
+            filtered_orig_node_type_ids = node_data[:,[5]][idx] # extract orig_type_node_id here
+            filtered_orig_node_type_ids = np.concatenate(filtered_orig_node_type_ids) 
+
+            if (node_type_name +'/feat' in node_features) and (node_features[node_type_name+'/feat'].shape[0] > 0): 
+                send_node_features[node_type_name+'/feat'] = node_features[node_type_name+'/feat'][filtered_orig_node_type_ids]
+            #else: 
+            #    send_node_features[node_type_name+'/feat'] = None
+
+        #accumulate subset of node_features targetted for part-id rank
+        node_features_rank_lst.append(send_node_features)
+
+    #send data
+    output_list = [None]
+    start = timer ()
+    dist.scatter_object_list(output_list, node_features_rank_lst, src=0)
+    end = timer ()
+
+    print( 'Rank: ', rank, ', Done sending Node Features to: ', part_id, ' in: ', timedelta(seconds = end - start))
+        
+
+def send_data(rank, node_data, node_features, edge_data, metis_partitions, ntypes_map): 
+    """
+    Wrapper function to send graph data to non-rank-0 processes.
+
+    Parameters:
+    -----------
+    rank : rank of the process
+    ranks_list : list of unique ranks/partition-ids
+    node_data : node_data, augmented, from xxx_nodes.txt file
+    node_features : node_features, data from the node_feats.dgl
+    edge_data : edge_data, augmented, from xxx_edges.txt file
+    metis_partitions : orig_node_id -> partition_id mappings as defined by METIS
+    ntype_map : dictionary of mappings between ntype_name -> ntype_id
+    """
+
+    partition_ids = np.unique(list(metis_partitions.values()))
+    partition_ids.sort ()
+    print( 'Rank: ', rank, ', Unique partitions: ', partition_ids)
+
+    send_node_data(rank, node_data, partition_ids)
+    send_edge_data(rank, edge_data, partition_ids) 
+    send_node_features(rank, node_data, node_features, partition_ids, ntypes_map)
+
+
+def recv_data(rank, dimensions, dtype): 
```

Comment:
can you use `shape` instead of `dimensions`? the word dimension can have different meanings in ML program.